dbms/src/Flash/Mpp/tests/gtest_aborted_mpp_gather_cache.cpp

```diff
@@ -0,0 +1,59 @@
+// Copyright 2022 PingCAP, Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <Flash/Mpp/MPPTaskManager.h>
+#include <TestUtils/TiFlashTestBasic.h>
+#include <gtest/gtest.h>
+
+
+namespace DB
+{
+namespace tests
+{
+class TestAbortedMPPGatherCache : public testing::Test
```

Comment:
It looks like we can construct the case described in pr in `gtest_compute_server.cpp`.
```
TEST_F(ComputeServerRunner, xxx)
try
{
    WRAP_FOR_SERVER_TEST_BEGIN
    startServers(4);
    {
        auto [query_id, holder] = prepareMPPStreams(context
                                                     .scan("test_db", "test_table_1")
                                                     .aggregation({Max(col("s1"))}, {col("s2"), col("s3")})
                                                     .project({"max(s1)"}));
        MockComputeServerManager::instance().cancelQuery(query_id);
       EXPECT_TRUE(!assertQueryActive(query_id)); 
       holder.runMPPTasks();
        EXPECT_TRUE(assertQueryCancelled(query_id));
    }
    WRAP_FOR_SERVER_TEST_END
}
CATCH
```

Reply:
just an option

Reply:
OK, I've added a test in `gtest_compute_server.cpp`